Dynatrace/dynatrace-operator
# auto update release branch ondemand e2e

## Description

This PR will add new workflow for update release branch when new branch is created similar we do with renovate, but
inside ondemand pipeline.

[K8S-9688](https://dt-rnd.atlassian.net/browse/K8S-9688)

> Sidenote:
> I have introduced a new yaml lib because Pyyaml doesn't support new standards and cannot parse properly GH action yaml files.

## How can this be tested?

1. Run git command:
`git branch -r --list 'origin/release-*' | sort --version-sort | tail -n 1| tr -d ' ' > release-branches.txt`
2. Update https://github.com/dynatrace/dynatrace-operator/blob/959ec573813a8a02f4b50e74c32dbcebb62fc456/.github/workflows/e2e-tests-ondemand.yaml#L12 to `release-1.0`
3. `python hack/build/ci/update-e2e-ondemand-pipeline.py release-branches.txt .github/workflows/e2e-tests-ondemand.yaml`
4. Make sure default branch in `.github/workflows/e2e-tests-ondemand.yaml`: 
```
...
  workflow_dispatch:
    inputs:
      target:
        description: Target branch to run E2E tests over
        required: true
        default: release-1.3
...
```
